gc_registry/measurement/routes.py

```diff
@@ -1,17 +1,61 @@
 # Imports
 from esdbclient import EventStoreDBClient
-from fastapi import APIRouter, Depends
+from fastapi import APIRouter, Depends, HTTPException
 from sqlmodel import Session
 
 from gc_registry.core.database import db, events
 from gc_registry.measurement import models
+from gc_registry.measurement.services import parse_measurement_json
 
 # Router initialisation
 router = APIRouter(tags=["Measurements"])
 
 ### Device Meter Readings ###
 
 
+@router.post("/submit_readings", response_model=models.MeasurementSubmissionResponse)
+def submit_readings(
```

Comment:
What does providing the `measurement_json` to the front-end look like for the UI? It looks like in FastAPI you might be able to skip some of this with something like the following:

```
@app.post("/upload")
def upload(file: UploadFile = File(...)):
    csvReader = csv.DictReader(codecs.iterdecode(file.file, 'utf-8'))
    data = {}
    for rows in csvReader:             
        key = rows['Id']  # Assuming a column named 'Id' to be the primary key
        data[key] = rows  
    
    file.file.close()
    return data

```

Reply:
Yeah I saw that bit of code as well, for the front end that will likely be the approach but for now it's just for the demo notebook. It'll be provide a path to the CSV to the serialize_meter_reasings function that can then just be passed directly as `content` to the endpoint